File: 0036-valid-sudoku/0036-valid-sudoku.ts
function isValidSudoku(board: string[][]): boolean {
    const seen = new Set();
    
    for (let i = 0; i < 9; i++) {
        for (let j = 0; j < 9; j++) {
            const currentVal = board[i][j];
            
            if (currentVal !== '.') {
                const boxIndex = Math.floor(i / 3) * 3 + Math.floor(j / 3);
                
                const rowKey = `${currentVal} in row ${i}`;
                const colKey = `${currentVal} in col ${j}`;
                const boxKey = `${currentVal} in box ${boxIndex}`;
                
                if (seen.has(rowKey) || seen.has(colKey) || seen.has(boxKey)) {
                    return false;
                }
                
                seen.add(rowKey);
                seen.add(colKey);
                seen.add(boxKey);
            }
        }
    }
    return true;
};

File: 0036-valid-sudoku/0036-valid-sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        Set<String> seen = new HashSet<>();

        for(int i = 0; i < 9; i++) {
            for(int j = 0; j < 9; j++) {
                char currVal = board[i][j];

                if(currVal != '.') {
                    int boxIdx = (i/3) * 3 + (j/3);

                     if (!seen.add(currVal + " in row " + i) ||
                        !seen.add(currVal + " in col " + j) ||
                        !seen.add(currVal + " in box " + boxIdx)) {
                        return false;
                    }
                }
            }
        } 
        return true;
    }
}